seismicpro/src/seismic_batch.py

```diff
@@ -693,14 +709,46 @@ def drop_zero_traces(self, index, src, num_zero, **kwargs):
         -------
             : SeismicBatch
             Batch without dropped traces.
+
+        Raises
+        ------
+        ValueError : if `src` has no sorting and batch index is FieldIndex
+        ValueError : if any component in batch has sorting different from `src`
+
+        Note
+        ----
+        This action creates new instance of SeismicBatch with new index
+        instance.
         """
         _ = kwargs
+        sorting = self.meta[src]['sorting']
+        if sorting is None and isinstance(self.index, FieldIndex):
```

Comment:
```suggestion
        if sorting is None and not isinstance(self.index, TraceIndex):
```
I think in case we use, for example, `Common Midpoint Index` or `Common Receiver index` we must still care about sorting